### PR TITLE
Sexy search

### PR DESCRIPTION
**Pull request by annatraussnig**
- Remove least used filters and replace them with a tab interface
- Remove tracking ID (@patrickvenetz told me we didn't need this anymore)
- Better lexical highlighting
- Move state management in URL (except for fetchMore logic)
- Split code in smaller, more manageable parts
- Update and show results count in empty state
- Add cheatsheet for more advanced search commands

### Empty State
Screenshot 2019-12-17 at 17 09 25
Screenshot 2019-12-17 at 17 09 33

### Search Example
Screenshot 2019-12-05 at 16 42 03

### Highlighting Examples
Screenshot 2019-12-05 at 16 42 28
Screenshot 2019-12-05 at 16 42 40

### CODE REVIEW

File: components/Search/Filters.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    if (!search) return <EmptyState />
+
+    const { totalCount, aggregations } = search
+    if (totalCount === 0 || !aggregations) return <EmptyState />
+
+    return (
+      <ul {...styles.list}>
+        {SUPPORTED_FILTERS.map((filter, key) => {
+          const agg = findAggregation(aggregations, filter)
+          // TODO: handle case where agg is undefined (maybe in backend?)
```
Comment (annatraussnig) [inline components/Search/Filters.js]: @patrickvenetz as soon as you have time taking a look at this we can consider realising this chunk of work :)

File: components/Search/Form.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    updateUrlQuery,
+    resetUrl,
+    dataAggregations,
+    t,
+    searchQuery,
+    setSearchQuery
+  }) => {
+    const [focusRef, setFocusRef] = useState(null)
+    const [formValue, setFormValue] = useState(urlQuery)
+    const [slowFormValue] = useDebounce(formValue, 200)
```
Comment (annatraussnig) [inline components/Search/Form.js]: This is for the backend not to get a crazy amount of requests through the withAggregations component, while keeping the frontend responsive

File: lib/utils/helpers.js [hunk 3]
```diff
@@ -14,3 +14,5 @@ export const intersperse = (list, separator) => {
 export const getRandomInt = limit => Math.floor(Math.random() * limit)
 
 export const getElementFromSeed = (list, seed) => list[seed % list.length]
+
+export const findByKey = (list, key, value) => list.find(e => e[key] === value)
```
Comment (tpreusse) [inline lib/utils/helpers.js]: This abstract increases the complexity of the code and make it harder to understand what is going on. Going to remove it.

File: components/Search/withSearchRouter.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      return updateURL({
+        [SORT_KEY_PARAM]: isDefault ? undefined : sort.key,
+        [SORT_DIRECTION_PARAM]: isDefault ? undefined : sort.direction
+      })
+    }
+
+    const resetUrl = () => Router.pushRoute('search')
+
+    const cleanupUrl = () =>
+      Router.pushRoute(
```
Comment (tpreusse) [inline components/Search/withSearchRouter.js]: `replaceRoute` is better here – replacing the dirty url from the browser history

**Pull request merged**